packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -482,9 +470,9 @@ export default class InternalModel {
     }
 
     // move to an empty never-loaded state
+    this.updateRecordArrays();
```

Comment:
Why did we have to move this? The update now might not have the correct state?

Reply:
I moved it when trying to determine if it's order was causing an issue in M3 (it wasn't). But I also realized it's current order is problematic as we notify observers on the record-array that the record has changed before we actually settle into the state we are moving to.